adapters/liftoff/liftoff.go

```diff
@@ -0,0 +1,148 @@
+			// therefore, their non-pointer values are not shared memory and are safe to modify.
+			imp.BidFloorCur = SupportedCurrency
+			imp.BidFloor = convertedValue
+		}
+
+		var impExt liftoffImpressionExt
+		if err := json.Unmarshal(imp.Ext, &impExt); err != nil {
+			errs = append(errs, fmt.Errorf("failed unmarshalling imp ext (err)%s", err.Error()))
+			continue
+		}
+
+		// get placement_reference_id & pub_app_store_id
+		var bidderImpExt openrtb_ext.ImpExtLiftoff
+		if err := json.Unmarshal(impExt.Bidder, &bidderImpExt); err != nil {
+			errs = append(errs, fmt.Errorf("failed unmarshalling bidder imp ext (err)%s", err.Error()))
+			continue
+		}
+
+		bidderImpExt.BidToken = requestCopy.User.BuyerUID
+		impExt.Ext = bidderImpExt
+		if newImpExt, err := json.Marshal(impExt); err == nil {
+			imp.Ext = newImpExt
+		} else {
+			errs = append(errs, errors.New("failed re-marshalling imp ext"))
+			continue
+		}
+
+		imp.TagID = bidderImpExt.PlacementRefID
+		requestCopy.Imp = []openrtb2.Imp{imp}
+		// must make a shallow copy for pointers.
+		requestAppCopy := *request.App
+		requestAppCopy.ID = bidderImpExt.PubAppStoreID
+		requestCopy.App = &requestAppCopy
+
+		requestJSON, err := json.Marshal(requestCopy)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		requestData := &adapters.RequestData{
+			Method: "POST",
+			Uri:    a.Endpoint,
+			Body:   requestJSON,
+			Headers: http.Header{
+				"Content-Type":      []string{"application/json"},
+				"Accept":            []string{"application/json"},
+				"X-OpenRTB-Version": []string{"2.5"},
+			},
+		}
+
+		requests = append(requests, requestData)
+	}
+
+	return requests, errs
+}
+
+// MakeBids collect bid response from liftoff and change them into the form that Prebid Server can handle.
+func (a *adapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if responseData.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if responseData.StatusCode == http.StatusBadRequest {
+		err := &errortypes.BadInput{
+			Message: "Unexpected status code: 400. Bad request from publisher. Run with request.debug = 1 for more info.",
+		}
+		return nil, []error{err}
+	}
+
+	if responseData.StatusCode != http.StatusOK {
+		err := &errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info.", responseData.StatusCode),
+		}
+		return nil, []error{err}
+	}
+
+	var response openrtb2.BidResponse
+	if err := json.Unmarshal(responseData.Body, &response); err != nil {
+		return nil, []error{err}
+	}
+
+	var errs []error
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(request.Imp))
+	bidResponse.Currency = response.Cur
+	for _, seatBid := range response.SeatBid {
+		for _, temp := range seatBid.Bid {
+			bid := temp // avoid taking address of a for loop variable
```

Comment:
this is correct but you might wanna consider the following code to avoid creating a new temp variable - 
```
for i := range seatBid.Bid {
    b := &adapters.TypedBid{
				Bid:     &seatBid.Bid[i],
				BidType: openrtb_ext.BidTypeVideo,
			}
       // more code

}
```

Reply:
Done optimizing